genet/core.py

```diff
@@ -432,6 +436,100 @@ def links_on_spatial_condition(self, region_input, how='intersect'):
             # is assumed to be hex
             return self._find_link_ids_on_s2_geometry(gdf, how, region_input)
 
+    def subnetwork(self, links: Union[list, set], services: Union[list, set] = None):
+        """
+        Subset a Network object using a collection of link IDs and (optionally) service IDs
+        :param links: Link IDs to be retained in the new Network
+        :param services: optional, collection of service IDs in the Schedule for subsetting.
+        :return: A new Network object that is a subset of the original
+        """
+        logging.info('Subsetting a Network will likely result in a disconnected network graph. A cleaner will be ran '
+                     'that will remove links to make the resulting Network strongly connected for modes: '
+                     'car, walk, bike.')
+        subnetwork = Network(epsg=self.epsg)
+        if self.schedule:
+            if services:
+                logging.info(
+                    f'Schedule will be subsetted using given services: {services}. Links pertaining to their '
+                    'network routes will also be retained.')
+                subschedule = self.schedule.subschedule(services)
+                routes = subschedule.route_attribute_data(keys=['route'])
+                links = set(links) | set(np.concatenate(routes['route'].values))
+                subnetwork.schedule = subschedule
+        subnetwork.graph = self.subgraph_on_link_conditions(conditions={'id': links})
+        subnetwork.link_id_mapping = {k: v for k, v in self.link_id_mapping.items() if k in links}
+        for mode in {'car', 'walk', 'bike'}:
+            if not subnetwork.is_strongly_connected(modes=mode):
+                subnetwork.retain_n_connected_subgraphs(n=1, mode=mode)
+
+        # TODO Inherit and subset Auxiliary files
+
+        logging.info('Subsetted Network is ready - do not forget to validate and visualise your subset!')
+        return subnetwork
+
+    def subnetwork_on_spatial_condition(self, region_input, how='intersect'):
+        """
+        Subset a Network object using a spatial bound
+        :param region_input:
+            - path to a geojson file, can have multiple features
+            - string with comma separated hex tokens of Google's S2 geometry, a region can be covered with cells and
```

Comment:
Lovely to see so many options. I imagine geojson will be v common, but S2 support is 🆒